### ISSUE TITLE
Verify incomplete submissions behavior

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#3084]
Currently incomplete submissions appear in the "All Active" list (e.g. for Editors).
In previous releases of OJS, incomplete submissions would only appear for the author who had started the submission.
If this was an intentional change, there will likely need to be some policy changes to permit access to incomplete submissions by users other than those who began submitting.
If it was an unintentional change, ensure that incomplete submissions are only listed to the users who began them (I suggest this approach).

**Comment by t4x0n** [comment]
Hello @asmecher, I want to tell you that I liked this way of seeing the incomplete submission, for the following reasons:

- A few days ago a user made three attempts to send and only one was actually sent or successfully, but two more were left as incomplete, that as Admin or Chief Editor I could delete them
- I was able to help a user to complete her submission because she had doubts about how to do it
- I could see two submissions where the "author" uploaded images in the abstract that said "this page was hacked by..." and with this, I could see that the user only had bad intentions and I could block it.

All the above thanks to the possibility of seeing incomplete submissions!  So, in my opinion, maybe is not a bad option give to the Journal Manager role the chance of seeing those incomplete submissions.

**Comment by NateWr** [comment]
@asmecher Yes, this was part intention, part inertia. Intentional for the reasons that @t4x0n describes. Inertial because they have `STATUS_QUEUED` and are technically in the submission stage, and it makes the filtering much simpler to just include them.

They appear at the bottom of the All Active list, so shouldn't be "in the way". Only Admins and Journal Managers have access to this list. Both also have access to the submission workflow for these submissions, so I'm not aware of any additional access issues.

**Comment by asmecher** [comment]
OK, thanks. I just wasn't sure if it was intentional behavior or not. I'll do a quick check to see whether there are access problems.

**Comment by asmecher** [comment]
If a submission is on step 2 (file upload) and the Editor follows a link to complete it, the file upload wizard generates an "Invalid uploader roles!" error. The submission wizard expects authors to upload documents, but the editor may not have an author role.

Two thoughts:
- The "uploader user ID" is an old concept and I'm not sure how/were it's used. It may be possible to audit and remove it entirely, and a lot of misery along with it.
- Rather than granting access for the Editor to participate in the author's submission area, it might be better to facilitate the "Log In As" feature for this purpose.

**Comment by NateWr** [comment]
I decided to allow editors to upload files for any submission in the wizard. It was straightforward and seems like a quicker route in for editors.

I also had a look at the `uploaderGroupIds`. I couldn't find any use of them outside this one case. I think I've removed everything without bad side effects, but please give it a close look @asmecher.

PRs:
https://github.com/pkp/pkp-lib/pull/3126
https://github.com/pkp/ojs/pull/1730
https://github.com/pkp/omp/pull/479

**Comment by asmecher** [comment]
Excellent work, @NateWr -- knowing this concept was still in the code was a source of stress. I suspect there are other good candidates for a similar treatment when it comes to reviewing the file upload and management code.

Please go ahead and merge -- now that we're prioritizing `ojs-stable-3_1_0` for the next release, it's actually a good time to merge semi-invasive changes.

**Comment by asmecher** [comment]
Oops, I see some areas that might be affected. Did you dig into `SubmissionFile::[gs]etUserGroupId`? That's where the `uploaderUserGroup` stuff gets stored -- it's in the DB, and has tendrils out into a lot of other functions.

**Comment by NateWr** [comment]
Yep, that was a deep rabbit hole. :laughing: I've removed all references I can find to submission file user group IDs. I've updated the PRs above and added one for `lensGalley`. Some notes:

- We used the uploader user group in anonymized uploaded file names for review. I've switched that to use the file genre. So instead of the default name being something like `Author, <original_file_name.doc>` it's now `Article Text <original_file_name.doc>`. Same for non-anonymized files, which are now `<author_name>, Article Text, <original_file_name.doc>`.
- I removed the `SubmissionFileManager::temporaryFileToSubmissionFile()` method. I couldn't find any uses of it and it didn't seem to have adverse effects. Sound right?
- When I ran the upgrade script in OJS it worked, but I got the following error, which didn't appear when I ran the same code in OMP. Maybe keep an eye out?

```
[data: dbscripts/xml/upgrade/3.1.1_update.xml]
ERROR: Upgrade failed: DB: Can't DROP 'user_group_id'; check that column/key exists
```
